samcli/commands/deploy/command.py

```diff
@@ -127,6 +128,7 @@
 @track_command
 @check_newer_version
 @print_cmdline_args
+@remove_stdin_tmpfile
```

Comment:
As @mildaniel suggested, please apply `@remove_stdin_tmpfile` to all commands that accept `--template-file`.